datalad/support/gitrepo.py

```diff
@@ -3532,6 +3532,16 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
                         else ('not a Git repository: %s', exc_str(e)),
                         logger=lgr)
                     continue
+                # This mirrors the result structure yielded for
+                # to_stage_submodules below.
+                yield get_status_dict(
+                    action='add',
+                    refds=self.pathobj,
+                    type='file',
+                    key=None,
+                    path=self.pathobj / ut.PurePosixPath(cand_sm),
```

Comment:
There could be places where yield a pathlib instance would be a problem. But I have a long-standing plan to address this at the main result handling loop... ;-)

Reply:
Sounds good (and this shouldn't cause any new problems since other places in gitrepo already yield pathlib objects).